src/Arcus.BackgroundJobs.CloudEvents/CloudEventMessageRouter.cs

```diff
@@ -30,17 +34,33 @@ protected override bool TryDeserializeToMessageFormat(string message, Type messa
             
             try
             {
+#if NET6_0
                 if (messageType == typeof(CloudEvent))
                 {
-                    CloudEvent cloudEvent = JsonEventFormatter.DecodeStructuredEvent(Encoding.UTF8.GetBytes(message));
-                    
-                    result = cloudEvent; 
+                    Logger.LogTrace("Deserialize incoming message as 'CloudEvent'...");
+                    CloudEvent cloudEvent = CloudEvent.Parse(BinaryData.FromString(message));
+                    Logger.LogTrace("Deserialized incoming message as 'CloudEvent'");
+
+                    result = cloudEvent;
+                    return true;
+                } 
+#endif
+
+                if (messageType == typeof(OldCloudEvent))
+                {
+                    Logger.LogWarning("Message handler uses old 'CloudNative.CloudEvents.CloudEvent' message type, please use the 'Azure.Messaging.CloudEvent' message type instead");
```

Comment:
Are these warnings really something we want to have in our logs for every event that is received / handled ?

Reply:
Maybe trace then?

Reply:
Just to be sure:  it is the producer of the event that decides which type of event (CloudNative.CloudEvent or Azure.CloudEvent) that must be used ?

Reply:
I think that's an architectural question. I think it is not necessarily because the original event producer published a `CloudEvent`, that we would get a `CloudEvent`. Maybe some integration conversion is involved. But, normally, I would say: yes, the producer creates and publishes the event and so decides which type it is.

Reply:
Well, in that case I don't think it makes sense that the handler of the raised events logs a warning, as the one responsible for the handler might not be the same person / team as the one responsible for the system that raises the event.
To avoid log cluttering, I would not log this as a warning.  But that's my idea. :) Maybe other people have different opinions. @pim-simons @gverstraete  ?

Reply:
> Well, in that case I don't think it makes sense that the handler of the raised events logs a warning, as the one responsible for the handler might not be the same person / team as the one responsible for the system that raises the event.

In this cause, though, the warning is about the message type of the message handler. So it is the person that receives the event that needs to change their message handler from the old to the new cloud event.

Reply:
Ok, then it is ok for me to have this as a warning.